e2e/src/unmount.test.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+import App from './App';
+
+enzyme.configure({ adapter: new Adapter() });
+
+describe('Given App is rendered and unmounted before final debounced render', () => {
+    let wrapper;
+
+    beforeEach(() => {
+        jest.useFakeTimers();
+        wrapper = enzyme.mount(<App />);
+    });
+
+    it('does not throw an unmounted component error', async () => {
```

Comment:
thanks for adding a test! 

Would you mind moving it to App.test.js? given the limited amount of tests and the code duplication in the setup/teardown I'd rather have the 2 tests collocated.

Also, it seems that the test doesn't fail if the error is logged, meaning that even without the fix the test would still pass.
We should instead spy on console.error, schedule an unmount at 90ms and test after triggering the timers if an error was logged.
 
Something like this:

```
    describe('When unmounts before a scheduled call to setState', () => {
        beforeEach(() => {
            jest.spyOn(global.console, 'error');
            setTimeout(() => wrapper.unmount(), 90);
            jest.runAllTimers();
        });

        it(`Doesn't log an error`, () => {
            expect(global.console.error).not.toHaveBeenCalled();
        });
    });
```

could you update your PR with the change?

Thanks!